lib/Spot/Entity.php

```diff
@@ -137,6 +139,18 @@ public function data($data = null, $modified = true)
                 $v = $this->__get($k, $v);
             }
 
+            foreach (self::$relationFields[$entityName] as $relationField) {
+                $relation = $this->relation($relationField);
+
+                if ($relation instanceof Entity\Collection) {
+                    $data[$relationField] = $relation->toArray();
+                }
+
+                if ($relation instanceof EntityInterface) {
```

Comment:
You should combine this and the previous `if` statement - both the collection and the entity have `toArray` methods, and they are both relations :)